File: Basic_Hash_Map.py
def main():
    hashmap = {}
    
    while True:
        print("Menu:")
        print("1. Insert key-value pair")
        print("2. Look up value by key")
        print("3. Delete key-value pair")
        print("4. Display the hash map")
        print("5. Quit")
        choice = input("Enter your choice: ")
        
        if not choice.isdigit():
            print("Invalid input. Please enter a valid option.")
            continue
        
        choice = int(choice)

        if choice == 1:
            key = input("Enter key: ")
            value = input("Enter value: ")
            try:
                hashmap[key] = int(value)
            except ValueError:
                print("Invalid input. Please enter a valid value.")
        elif choice == 2:
            key = input("Enter key to look up its value: ")
            value = hashmap.get(key, -1)
            if value != -1:
                print(f"Value for key '{key}' is {value}")
            else:
                print(f"Key '{key}' not found in the hash map")
        elif choice == 3:
            key = input("Enter key to delete: ")
            if key in hashmap:
                del hashmap[key]
                print(f"Key '{key}' deleted from the hash map")
            else:
                print(f"Key '{key}' not found in the hash map")
        elif choice == 4:
            print("Hash Map Contents:")
            for key, value in hashmap.items():
                print(f"Key: {key}, Value: {value}")
        elif choice == 5:
            break
        else:
            print("Invalid choice. Please select a valid option.")

if __name__ == "__main__":
    main()

File: Basic_Hash_Map.java
import java.util.HashMap;
import java.util.Scanner;

public class HashMapExample {
    public static void main(String[] args) {
        HashMap<String, Integer> hashMap = new HashMap<>();
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("Menu:");
            System.out.println("1. Insert key-value pair");
            System.out.println("2. Look up value by key");
            System.out.println("3. Delete key-value pair");
            System.out.println("4. Quit");
            System.out.print("Enter your choice: ");

            int choice = scanner.nextInt();
            scanner.nextLine();

            switch (choice) {
                case 1:
                    System.out.print("Enter key: ");
                    String key = scanner.nextLine();
                    System.out.print("Enter value: ");
                    int value = scanner.nextInt();
                    hashMap.put(key, value);
                    break;
                case 2:
                    System.out.print("Enter key to look up its value: ");
                    key = scanner.nextLine();
                    if (hashMap.containsKey(key)) {
                        System.out.println("Value for key '" + key + "' is " + hashMap.get(key));
                    } else {
                        System.out.println("Key '" + key + "' not found in the hash map");
                    }
                    break;
                case 3:
                    System.out.print("Enter key to delete: ");
                    key = scanner.nextLine();
                    if (hashMap.containsKey(key)) {
                        hashMap.remove(key);
                        System.out.println("Key '" + key + "' deleted from the hash map");
                    } else {
                        System.out.println("Key '" + key + "' not found in the hash map");
                    }
                    break;
                case 4:
                    scanner.close();
                    return;
                default:
                    System.out.println("Invalid choice. Please select a valid option.");
            }
        }
    }
}
